internal/locate/region_cache.go

```diff
@@ -398,6 +398,33 @@ func (r *Region) isValid() bool {
 	return r != nil && !r.checkNeedReload() && r.checkRegionCacheTTL(time.Now().Unix())
 }
 
+type regionIndexMu struct {
+	sync.RWMutex
+	regions        map[RegionVerID]*Region // cached regions are organized as regionVerID to region ref mapping
+	latestVersions map[uint64]RegionVerID  // cache the map from regionID to its latest RegionVerID
+	sorted         *SortedRegions          // cache regions are organized as sorted key to region ref mapping
+}
+
+func newRegionIndexMu(btreeDegree int, rs []*Region) *regionIndexMu {
+	r := &regionIndexMu{}
+	r.regions = make(map[RegionVerID]*Region)
+	r.latestVersions = make(map[uint64]RegionVerID)
+	r.sorted = NewSortedRegions(btreeDegree)
+	for _, region := range rs {
+		r.insertRegionToCache(region)
+	}
+	return r
+}
+
+func (mu *regionIndexMu) refresh(btreeDegree int, r []*Region) {
```

Comment:
I guess we don't need to `btreeDegree` argument as it is a constant that is always 32.

Reply:
ok, I will modify it later